simd/src/Kokkos_Simd_AVX512.hpp

```diff
@@ -0,0 +1,928 @@
+      _mm512_and_epi64(_mm512_set1_epi64(0x7FFFFFFFFFFFFFFF),
+                       reinterpret_cast<__m512i>(rhs))));
+}
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<double, simd_abi::avx512_fixed_size<8>> sqrt(
+    simd<double, simd_abi::avx512_fixed_size<8>> const& a) {
+  return simd<double, simd_abi::avx512_fixed_size<8>>(
+      _mm512_sqrt_pd(static_cast<__m512d>(a)));
+}
+
+#ifdef __INTEL_COMPILER
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<double, simd_abi::avx512_fixed_size<8>> cbrt(
+    simd<double, simd_abi::avx512_fixed_size<8>> const& a) {
+  return simd<double, simd_abi::avx512_fixed_size<8>>(
+      _mm512_cbrt_pd(static_cast<__m512d>(a)));
+}
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<double, simd_abi::avx512_fixed_size<8>> exp(
+    simd<double, simd_abi::avx512_fixed_size<8>> const& a) {
+  return simd<double, simd_abi::avx512_fixed_size<8>>(
+      _mm512_exp_pd(static_cast<__m512d>(a)));
+}
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<double, simd_abi::avx512_fixed_size<8>> log(
+    simd<double, simd_abi::avx512_fixed_size<8>> const& a) {
+  return simd<double, simd_abi::avx512_fixed_size<8>>(
+      _mm512_log_pd(static_cast<__m512d>(a)));
+}
+
+#endif
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<double, simd_abi::avx512_fixed_size<8>> fma(
+    simd<double, simd_abi::avx512_fixed_size<8>> const& a,
+    simd<double, simd_abi::avx512_fixed_size<8>> const& b,
+    simd<double, simd_abi::avx512_fixed_size<8>> const& c) {
+  return simd<double, simd_abi::avx512_fixed_size<8>>(
+      _mm512_fmadd_pd(static_cast<__m512d>(a), static_cast<__m512d>(b),
+                      static_cast<__m512d>(c)));
+}
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<double, simd_abi::avx512_fixed_size<8>> max(
+    simd<double, simd_abi::avx512_fixed_size<8>> const& a,
+    simd<double, simd_abi::avx512_fixed_size<8>> const& b) {
+  return simd<double, simd_abi::avx512_fixed_size<8>>(
+      _mm512_max_pd(static_cast<__m512d>(a), static_cast<__m512d>(b)));
+}
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<double, simd_abi::avx512_fixed_size<8>> min(
+    simd<double, simd_abi::avx512_fixed_size<8>> const& a,
+    simd<double, simd_abi::avx512_fixed_size<8>> const& b) {
+  return simd<double, simd_abi::avx512_fixed_size<8>>(
+      _mm512_min_pd(static_cast<__m512d>(a), static_cast<__m512d>(b)));
+}
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<double, simd_abi::avx512_fixed_size<8>> condition(
+    simd_mask<double, simd_abi::avx512_fixed_size<8>> const& a,
+    simd<double, simd_abi::avx512_fixed_size<8>> const& b,
+    simd<double, simd_abi::avx512_fixed_size<8>> const& c) {
+  return simd<double, simd_abi::avx512_fixed_size<8>>(
+      _mm512_mask_blend_pd(static_cast<__mmask8>(a), static_cast<__m512d>(c),
+                           static_cast<__m512d>(b)));
+}
+
+template <>
+class const_where_expression<simd_mask<double, simd_abi::avx512_fixed_size<8>>,
+                             simd<double, simd_abi::avx512_fixed_size<8>>> {
```

Comment:
https://en.cppreference.com/w/cpp/experimental/simd/const_where_expression also has `operator-` here.

Reply:
Yes, I would be happy to add them if demand is indicated. This pretty much just boils down to limited time.